plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java

```diff
@@ -132,6 +132,8 @@ public void testEnforcedCooldownPeriod() throws IOException {
             }
         })));
 
+        // Master failover to clear RepositoryData cache
```

Comment:
I think I would prefer an explicit undocumented setting to disable the cache. This might also turn out to be useful if we see any issues with this new functionality

Reply:
Yea that's much nicer indeed :) added that setting now and used it in tests.